chain/epoch-manager/src/adapter.rs

```diff
@@ -644,6 +649,14 @@ impl EpochManagerAdapter for EpochManagerHandle {
         self.get_shard_layout(&epoch_id)
     }
 
+    fn get_shard_layout_from_protocol_version(
+        &self,
+        protocol_version: ProtocolVersion,
+    ) -> Result<ShardLayout, EpochError> {
+        let epoch_manager = self.read();
+        Ok(epoch_manager.get_epoch_config(protocol_version).shard_layout)
```

Comment:
Can we change `get_shard_layout_from_protocol_version` to return a `ShardLayout`?

The version in `kv-runtime` shouldn always succeed